dotnet/src/Functions/Functions.Prompty/Extensions/PromptyKernelExtensions.cs

```diff
@@ -64,4 +65,58 @@ public static KernelFunction CreateFunctionFromPrompty(
             promptTemplateFactory ?? KernelFunctionPrompty.s_defaultTemplateFactory,
             kernel.LoggerFactory);
     }
+
+    /// <summary>
+    /// Create a <see cref="KernelFunction"/> from a prompty template file.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="promptyFilePath">Path to the file containing the Prompty representation of a prompt based <see cref="KernelFunction"/>.</param>
+    /// <param name="fileProvider">The representation of the file system to use to retrieve the prompty file. Defaults to <see cref="PhysicalFileProvider"/> scoped to the current directory.</param>
+    /// <param name="promptTemplateFactory">
+    /// The <see cref="IPromptTemplateFactory"/> to use when interpreting the prompt template configuration into a <see cref="IPromptTemplate"/>.
+    /// If null, a <see cref="AggregatorPromptTemplateFactory"/> will be used with support for Liquid and Handlebars prompt templates.
+    /// </param>
+    /// <returns>The created <see cref="KernelFunction"/>.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="kernel"/> is null.</exception>
+    /// <exception cref="ArgumentNullException"><paramref name="promptyFilePath"/> is null.</exception>
+    /// <exception cref="ArgumentException"><paramref name="promptyFilePath"/> is empty or composed entirely of whitespace.</exception>
+    public static KernelFunction CreateFunctionFromPromptyFile(
+        this Kernel kernel,
+        string promptyFilePath,
+        IFileProvider fileProvider,
+        IPromptTemplateFactory? promptTemplateFactory = null)
+    {
+        Verify.NotNull(kernel);
+        Verify.NotNullOrWhiteSpace(promptyFilePath);
+        Verify.NotNull(fileProvider);
+
+        var fileInfo = fileProvider.GetFileInfo(promptyFilePath);
+        return CreateFunctionFromPromptyFile(kernel, fileInfo, promptTemplateFactory);
+    }
+
+    /// <summary>
+    /// Create a <see cref="KernelFunction"/> from a prompty template file.
+    /// </summary>    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="fileInfo">The file containing the Prompty representation of a prompt based <see cref="KernelFunction"/>.</param>
```

Comment:
Small formatting fix:
```suggestion
    /// <summary>
    /// Create a <see cref="KernelFunction"/> from a prompty template file.
    /// </summary>    
    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
    /// <param name="fileInfo">The file containing the Prompty representation of a prompt based <see cref="KernelFunction"/>.</param>
```